modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/OsgiAssetManager.java

```diff
@@ -197,6 +238,22 @@ public void setAssetStore(AssetStore assetStore) {
     this.assetStore = assetStore;
   }
 
+  public void addRemoteAssetStore(RemoteAssetStore assetStore) {
```

Comment:
Since we can (in theory) count on OSGI enforcing the dependency graphs between the various services, we know that the delegate already has to exist for this method to even be called, correct?  Karaf won't be attempting to bind new stores before the activate() method has returned.  Thus, we don't need to synchronize this, especially if we rip out the delegate protection that's present in this method.

At least, that's my read of the spec...

Reply:
I would agree if the remote store dependency were 'static'. But it is 'dynamic' (as it should be) and so the 'bind' method can be called after the class has been instantiated, but before activate(). With this change, it's possible to get a null pointer.

Reply:
Huh, that's a dumb bug in the spec IMO.  Regardless, how would synchronizing the method help?  It would prevent multiple threads from adding assets stores at the same time, but it wouldn't stop the null pointer.

Reply:
The previous version of the code had an intermediate set (remotes) where the remote stores were added if the delegate was not ready yet.

Reply:
Ah, I get it now.  Ok, added synchronization in relevant places!